blocks/extensions/saveload.py

```diff
@@ -119,49 +120,28 @@ class LoadFromDump(TrainingExtension):
 
     """
     def __init__(self, state_path, **kwargs):
-        super(LoadFromDump, self).__init__(**kwargs)
-        self.manager = MainLoopDumpManager(state_path)
+        super(Load, self).__init__(**kwargs)
+        self.state_path = state_path
+
+    def load(self):
+        with open(self.state_path, "rb") as source:
+            return load(source)
+
+    def load_to(self, main_loop):
+        loaded_main_loop = self.load()
```

Comment:
We could pickle the log and iteration state to separate files in the same zip archive, using the same persistent pickling approach. In general though, the situation covered by `LoadFromDump` that isn't covered by `Load` is quite specific, namely:

* The iteration state is picklable, but some other part of the main loop isn't
* The epoch is very long (otherwise the iteration state isn't very important)
* You don't care about the state of your extensions, algorithm, etc.

Otherwise you can just load the parameters and resume at the beginning of an epoch. So, although I agree that eventually this robustness would be a good idea, I think it's a relatively specific case, and the advantages of this new approach outweigh it IMHO.